diff --git a/arch/arm/boot/dts/sun7i-a20-bananapro.dts b/arch/arm/boot/dts/sun7i-a20-bananapro.dts
index e7af1b7..d7eac60 100644
--- a/arch/arm/boot/dts/sun7i-a20-bananapro.dts
+++ b/arch/arm/boot/dts/sun7i-a20-bananapro.dts
@@ -147,6 +147,25 @@
 	status = "okay";
 };
 
+&i2c3 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c3_pins_a>;
+	edt: edt-ft5x06@38 {
+		compatible = "edt,edt-ft5x06";
+		reg = <0x38>;
+		interrupt-parent = <&pio>;
+		interrupts = <7 9 IRQ_TYPE_EDGE_FALLING>;
+		wake-gpios = <7 7 GPIO_ACTIVE_LOW>;
+		pinctrl-names = "default";
+		touchscreen-size-x = <1024>;
+		touchscreen-size-y = <600>;
+		touchscreen-inverted-x;
+		touchscreen-swapped-x-y;
+	};
+};
+
+
 &ir0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&ir0_rx_pins_a>;
@@ -220,6 +239,11 @@
 		pins = "PH22";
 		function = "gpio_out";
 	};
+
+	edt_ft5x06_pins: edt_ft5x06_pins@0 {
+		pins = "PH7", "PH9";
+		function = "gpio_out";
+	};
 };
 
 &reg_usb1_vbus {
